config/plugin_printer.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+)
+
+var headers = []string{"Name", "Source(s)"}
+
+type pluginPrinter struct {
+	name   string
+	source string
+}
+
+type pluginNames []pluginPrinter
+
+func (p *pluginNames) String() string {
+	pluginMap := make(map[string][]string)
+	for _, plugin := range *p {
+		if _, ok := pluginMap[plugin.name]; !ok {
+			pluginMap[plugin.name] = make([]string, 0)
+		}
+		pluginMap[plugin.name] = append(pluginMap[plugin.name], plugin.source)
+	}
+
+	data := make([][]any, 0, len(pluginMap))
+	for name, sources := range pluginMap {
+		var nameCountStr string
+		if len(sources) > 1 {
+			nameCountStr = fmt.Sprintf("%s (%dx)", name, len(sources))
+		} else {
+			nameCountStr = name
+		}
+		data = append(data, []any{nameCountStr, sources})
+	}
+	// sort the data based on counts(i.e number of sources)
+	sort.Slice(data, func(i, j int) bool {
+		return len(data[i][1].([]string)) > len(data[j][1].([]string))
+	})
+
+	if len(data) == 0 {
+		return ""
+	}
+
+	return GetPluginTableContent(headers, data)
+}
+
+// GetPluginTableContent prints a bordered ASCII table.
+//
+// Inputs:
+//
+//	headers: slice of strings containing the headers of the table
+//	data: slice of slices of strings containing the data to be printed
+//
+// Reference: https://github.com/olekukonko/tablewriter
+func GetPluginTableContent(headers []string, data [][]any) string {
```

Comment:
I think using https://github.com/jedib0t/go-pretty here is the better alternative. :-)

Reply:
Yeah I'd rather add a dependency than stuffing in code here. :-)

Reply:
Done.